grakn/service/Session/TransactionService.py

```diff
@@ -165,7 +165,12 @@ def send(self, request):
         return response
 
     def close(self):
-        with self._queue.mutex: # probably don't even need the mutex
-            self._queue.queue.clear()
-        self._queue.put(None)
-        self._closed = True
+        if not self._closed:
+            with self._queue.mutex: # probably don't even need the mutex
+                self._queue.queue.clear()
+            self._queue.put(None)
+            self._closed = True
+            # force exhaust the iterator so `onCompleted()` is called on the server
+            # note: next() was throwing an error sometimes, but would be preferable to a loop
```

Comment:
can you copy/paste the error that is thrown by `next()` @flyingsilverfin ?

Reply:
just double checking it was this error not another one covering it up! Will do if it still fails

Reply:
Further investigation shows `next()` was giving "ERROR", but this was caused by a `StopIteration` exception that just needs to be caught. updated.